transmogrifier/sources/json/aardvark.py

```diff
@@ -124,6 +151,146 @@ def get_contributors(source_record: dict) -> list[timdex.Contributor]:
+        """Get values from source record for TIMDEX dates field."""
+        return (
+            cls._issued_dates(source_record)
+            + cls._coverage_dates(source_record)
+            + cls._range_dates(source_record, source_record_id)
+        )
+
+    @classmethod
+    def _issued_dates(cls, source_record: dict) -> list[timdex.Date]:
+        """Get values for issued dates."""
+        issued_dates = []
+        if "dct_issued_s" in source_record:
+            issued_dates.append(
+                timdex.Date(value=source_record["dct_issued_s"], kind="Issued")
+            )
+        return issued_dates
+
+    @classmethod
+    def _coverage_dates(cls, source_record: dict) -> list[timdex.Date]:
+        """Get values for coverage dates."""
+        coverage_dates = []
+        coverage_date_values = []
+        coverage_date_values.extend(source_record.get("dct_temporal_sm", []))
+        for date_value in [
+            str(date_value)
+            for date_value in source_record.get("gbl_indexYear_im", [])
+            if str(date_value) not in coverage_date_values
+        ]:
+            coverage_date_values.append(date_value)
+        for coverage_date_value in coverage_date_values:
+            coverage_dates.append(
+                timdex.Date(value=coverage_date_value, kind="Coverage")
+            )
+        return coverage_dates
+
+    @classmethod
+    def _range_dates(
+        cls, source_record: dict, source_record_id: str
+    ) -> list[timdex.Date]:
+        """Get values for issued dates."""
+        range_dates = []
+        for date_range_string in [
+            date_range_strings
+            for date_range_strings in source_record.get("gbl_dateRange_drsim", [])
+        ]:
+            date_range_values = cls.parse_solr_date_range_string(
+                date_range_string, source_record_id
+            )
+            range_dates.append(
+                timdex.Date(
+                    range=timdex.Date_Range(
+                        gte=date_range_values[0], lte=date_range_values[1]
+                    )
+                )
+            )
+        return range_dates
+
+    @classmethod
+    def parse_solr_date_range_string(
+        cls, date_range_string: str, source_record_id: str
+    ) -> tuple:
+        """Get a list of values from a Solr-formatted date range string.
+
+        Example:
+         - "[1943 TO 1946]"
+
+        Args:
+            date_range_string: Formatted date range string to parse.
+            source_record_id: The ID of the record containing the string to parse.
+        """
+        try:
+            matches = re.match(r"\[([0-9]{4}) TO ([0-9]{4})\]", date_range_string)
+            return matches.groups()  # type: ignore[union-attr]
+        except AttributeError:
+            message = (
+                f"Record ID '{source_record_id}': "
+                f"Unable to parse date range string '{date_range_string}'"
+            )
+            raise ValueError(message)
```

Comment:
I like this.  I was going to suggest that if `matches` is `None`, we just return an empty tuple instead of raising an error.  But on further reflection, I see that this method should only be called if `gbl_dateRange_drsim` is present, and _that_ should only have values in this pattern.  Therefore, raising an Exception for an inability to parse years from that feels accurate.  Good call!